### PR TITLE
test(channel): add unit tests for NativeChannel

### PR DESCRIPTION
**Pull request by Utkal059**
Add Vitest unit-test coverage for `NativeChannel`, which previously had none. Tests mock `fetch`, `FileReader`, and the native base64 encoder so the suite runs in a standard Node/Vitest environment without a browser or live gRPC-Web endpoint.

- Add `test/unit/channel/NativeChannel.js` (24 cases, 315 lines)
- Cover constructor: `_address` stored, `_isReady` defaults to `false`, custom `grpcDeadline` honoured
- Cover `close()`: confirmed no-op returning `undefined`
- Cover `_waitForReady()`: cached `_isReady` fast path; `https://` for remote and `http://` for `localhost` / `127.0.0.1`; past-deadline throws `Timeout`; `grpc-status` and `grpc-message` headers each mark the channel ready; HTTP 200 with no gRPC headers throws `Unavailable`; `AbortError` maps to `Timeout`; existing `GrpcServiceError` re-thrown unchanged; generic network error maps to `Unavailable`
- Cover `_createUnaryClient()`: `GrpcServiceError` propagated to callback; URL built as `https://<addr>/proto.<svc>/<method>` for remote and `http://...` for localhost; non-OK HTTP returns `HttpError`; both `application/octet-stream` and `application/grpc-web+proto` base64 prefixes decoded; unknown MIME prefix surfaces a descriptive `Error`; fully successful round-trip returns a `Uint8Array`

**Related issue(s)**:

N/A — coverage improvement, no linked issue.

**Notes for reviewer**:

The `FileReader` stub fires `load` / `loadend` via `queueMicrotask` rather than synchronously inside `readAsDataURL`. This matches real browser behaviour and is required because `NativeChannel.js` assigns `reader.onloadend` *after* calling `readAsDataURL(blob)` — a synchronous stub would fire before the handler is attached and every decode test would hang.

No production code changed; this PR is test-only.

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: test/unit/channel/NativeChannel.js [hunk 1]
```diff
@@ -0,0 +1,317 @@
+// SPDX-License-Identifier: Apache-2.0
+import { vi, describe, it, expect, beforeEach, afterEach } from "vitest";
```
Comment (venilinvasilev) [inline test/unit/channel/NativeChannel.js]: No need to import these

**Comment by venilinvasilev** [comment]
Hi @Utkal059, thanks for taking the time to contribute!
The test file should be at test/unit/NativeChannel.js rather than test/unit/channel/NativeChannel.js - we don't mirror the source directory structure in tests. The only permitted subdirectories are test/unit/node/ and test/unit/browser/ for environment-specific tests; everything else goes directly under test/unit/.

Sorry the original issue had the wrong path  it's been corrected now!

**Comment by Utkal059** [comment]
Thanks @venilinvasilev! Moved the file to test/unit/NativeChannel.js and removed the explicit Vitest globals import. Also updated the relative import paths accordingly. Force-pushed — please let me know if anything else needs adjusting!

**Comment by ivaylonikolov7** [comment]
Thank you for the help! 

You need to import `vi`. `task build` is failing.

**Comment by ivaylonikolov7** [comment]
closing this due to inactivity
